resources/js/sql.ts

```diff
@@ -1301,17 +1301,14 @@ function checkSavedQuery () {
     let key = Sql.getAutoSavedKey();
     let buttonGetAutoSavedQuery = $('#saved');
 
-    let isAutoSavedInLocalStorage = isStorageSupported('localStorage') && (typeof window.localStorage.getItem(key) === 'string');
+    if (isStorageSupported('localStorage') && typeof window.localStorage.getItem(key) === 'string') {
+        buttonGetAutoSavedQuery.prop('disabled', false);
     // @ts-ignore
-    let isAutoSavedInCookie = window.Cookies.get(key, { path: CommonParams.get('rootPath') });
-
-    if (isAutoSavedInLocalStorage || isAutoSavedInCookie) {
+    } else if (window.Cookies.get(key, { path: CommonParams.get('rootPath') })) {
         buttonGetAutoSavedQuery.prop('disabled', false);
-
-        return;
+    } else {
+        buttonGetAutoSavedQuery.prop('disabled', true);
     }
```

Comment:
It seems the `//@ts-ignore` is hiding a bug and the second argument to `Cookies.get()` is never used.
https://github.com/js-cookie/js-cookie/blob/ab3f67fc4fad88cdf07b258c08e4164e06bf7506/src/api.mjs#L50-L53
Should use `withAttributes` instead.
https://github.com/js-cookie/js-cookie/blob/ab3f67fc4fad88cdf07b258c08e4164e06bf7506/src/api.mjs#L89-L91

```suggestion
    const key = Sql.getAutoSavedKey();
    const haveAutoSavedQuery =
        (isStorageSupported('localStorage') && typeof window.localStorage.getItem(key) === 'string') ||
        window.Cookies.withAttributes({ path: CommonParams.get('rootPath') }).get(key);

    $('#saved').prop('disabled', !haveAutoSavedQuery);
```

Reply:
If that second parameter was never used nor needed, wouldn't the correct way be to just scrap the rootPath attribute altogether?

Reply:
Alright, it seems that GitHub updated the review UI at some point, it used to be good before but now it's incredibly bugged and shows the diff of your suggestion in relation to the original code and not to my modifications, plus the format is now all weird. Interesting how often development goes that you get the lightning in the bottle and make a great thing, but for some reason you then break it with updates.

Anyway, sorry for the rant, I think I got what you meant with your suggestion extracted.